### PR TITLE
:sparkles: Add MachineHealthCheck types

### PR DESCRIPTION
**Pull request by JoelSpeed**
**What this PR does / why we need it**:

This PR adds the required types for the Machine Health Checking proposal tracked in #1990 

- MachineHealthCheck
- MachineHealthCheckList
- MachineHealthCheckSpec
- UnhealthyCondition
- MachineHealthCheckStatus

### CODE REVIEW

File: api/v1alpha3/machinehealthcheck_types.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+// ANCHOR: MachineHealthCheckStatus
+
+// MachineHealthCheckStatus defines the observed state of MachineHealthCheck
+type MachineHealthCheckStatus struct {
+	// total number of machines counted by this machine health check
+	// +kubebuilder:validation:Minimum=0
+	ExpectedMachines *int `json:"expectedMachines"`
+
+	// total number of machines counted by this machine health check
```
Comment (enxebre) [inline api/v1alpha3/machinehealthcheck_types.go]: total number of *healthy* machines counted by this machine health check

Comment (JoelSpeed) [reply api/v1alpha3/machinehealthcheck_types.go]: Fixed! Good spot

**Comment by chuckha** [comment]
/uncc @randomvariable 
My fault, accidentally added you and it was the wrong PR.

**Comment by randomvariable** [comment]
Hah, I was just about to say that we can treat this completely independently.

File: api/v1alpha3/machinehealthcheck_types.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```
Comment (enxebre) [inline api/v1alpha3/machinehealthcheck_types.go]: 2020

**Comment by enxebre** [comment]
this lgtm as a first step so we can follow up with the reconciling logic, PTAL @detiber @ncdc

**Comment by ncdc** [comment]
FYI, I have a large PR review backlog. I'm getting through it as fast as I can. I apologize if it takes some time to get to this. Thanks for your patience.

**Comment by JoelSpeed** [comment]
Closing in favour of #2101 to keep all the work together (re https://github.com/kubernetes-sigs/cluster-api/pull/2101#issuecomment-576771109)

**Pull request closed without merging**